app/v3/controllers/SubmitFinalDeclarationController.scala

```diff
@@ -65,4 +71,40 @@ class SubmitFinalDeclarationController @Inject() (val authService: EnrolmentsAut
       requestHandler.handleRequest(rawData)
     }
 
+  private def updateNrs(nino: String, parsedRequest: SubmitFinalDeclarationRequest)(implicit ctx: RequestContext, ec: ExecutionContext): Unit = {
+    implicit val hc: HeaderCarrier = ctx.hc
+
+    retrieveCalculationDetails(parsedRequest).map {
+      case Left(_) =>
+        // details lookup failed, so as a fallback, just send the calculation ID
+        nrsProxyService.submit(nino, "itsa-crystallisation", parsedRequest.toNrsJson)
+
+      case Right(responseWrapper) =>
+        val details: RetrieveCalculationResponse = responseWrapper.responseData
+        nrsProxyService.submit(nino, "itsa-crystallisation", Json.toJson(details))
+    }
+  }
+
+  private def retrieveCalculationDetails(parsedRequest: SubmitFinalDeclarationRequest,
+                                         retry: Int = 0)(implicit ctx: RequestContext,
+                                                         ec: ExecutionContext): Future[ServiceOutcome[RetrieveCalculationResponse]] = {
+    import parsedRequest._
+
+    val retrieveRequest = RetrieveCalculationRequest(nino, taxYear, calculationId)
+    retrieveService.retrieveCalculation(retrieveRequest).flatMap {
+      case Right(result) =>
+        Future.successful(Right(result))
+      case Left(error) =>
+        if (retry <= 3) {
```

Comment:
```suggestion
        if (retry > 3) {
``` 
We need to update this because the retry will never be triggered otherwise